altapay.php

```diff
@@ -2034,7 +2047,7 @@ public function getTemplateVarInfos()
     {
         $cart = $this->context->cart;
         $currency = $this->getCurrencyForCart($cart);
-        $paymentMethods = Altapay_Models_Terminal::getActiveTerminalsForCurrency($currency->iso_code);
+        $paymentMethods = Altapay_Models_Terminal::getActiveTerminalsForCurrency($currency->iso_code, (int) $this->context->shop->id);
```

Comment:
You are validating the the shop_id in the function, why is it being casted now?

Reply:
Done